src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -261,6 +261,17 @@ public String getStashRepository() throws StashConfigurationException {
    */
   public String getStashPullRequestId() throws StashConfigurationException {
     String result = config.getPullRequestId();
+    
+    if(StringUtils.isNotBlank(config.getPullRequestBranch())) {
```

Comment:
This is the main part for https://github.com/AmadeusITGroup/sonar-stash/issues/92 - doing this in StashIssueReportingPostJob directly was not that easy (obscure junit test failures) but it would probably cleaner to move it there.